types/ember/index.d.ts

```diff
@@ -1138,11 +1138,7 @@ declare module 'ember' {
              * function returns true for. This method corresponds to `filter()` defined in
              * JavaScript 1.6.
              */
-            filter<S extends T>(
-                callbackfn: (value: T, index: number, array: T[]) => value is S,
-                thisArg?: any
-            ): S[];
-            filter(callbackfn: (value: T, index: number, array: T[]) => any, thisArg?: any): NativeArray<T>;
+            filter: GlobalArray<T>['filter'];
```

Comment:
This used to return `NativeArray`, but only in one overload, was that just wrong?

Reply:
`NativeArray<T>` (which extends `T[]`) would be more correct but I'm not sure we can use mapped types to convert the return type.

Reply:
You can't without Microsoft/TypeScript#6606